ui-admin/src/util/withStateResetOnEnvChange.test.tsx

```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import { render, screen } from '@testing-library/react'
+import { withStateResetOnEnvChange } from './withStateResetOnEnvChange'
+import { mockStudyEnvContext } from 'test-utils/mocking-utils'
+
+const MockComponent = ({ text }: { text: string }) => {
+  return <div data-testid="mock-component">{text}</div>
+}
+
+const WrappedComponent = withStateResetOnEnvChange(MockComponent)
+
+test('resets component state when studyEnvContext changes', () => {
+  const { rerender } = render(
+    <WrappedComponent studyEnvContext={{ ...mockStudyEnvContext(), currentEnvPath: 'mystudy/sandbox' }} text="sandbox"/>
+  )
+
+  expect(screen.getByTestId('mock-component')).toHaveTextContent('sandbox')
+
+  rerender(
```

Comment:
this test isn't testing the wrapper -- since the 'text' argument changes, the component will rerender anyway.  I think what you'll want to do is have the mock component have some state (e.g. a textbox) and then you userevent.type some characters to update the mock component state.  Then check that changing the studyenvironment resets the component state